src/Modules/ContentBlocks/Domain/ContentBlock/Command/CreateContentBlockHandler.php

```diff
@@ -5,38 +5,30 @@
 use ForkCMS\Core\Domain\MessageHandler\CommandHandlerInterface;
 use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlock;
 use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlockRepository;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Service\UpdateContentBlockWidget;
 use ForkCMS\Modules\Frontend\Domain\Block\Block;
 use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
 use ForkCMS\Modules\Frontend\Domain\Block\ModuleBlock;
 use ForkCMS\Modules\Internationalisation\Domain\Translation\TranslationKey;
+use ForkCMS\Modules\ContentBlocks\Frontend\Widgets\Detail as DetailWidget;
 
 final class CreateContentBlockHandler implements CommandHandlerInterface
 {
     public function __construct(
-        private readonly BlockRepository $blockRepository,
-        private readonly ContentBlockRepository $contentBlockRepository
+        private readonly ContentBlockRepository $contentBlockRepository,
+        private readonly UpdateContentBlockWidget $widget
     ) {
     }
 
     public function __invoke(CreateContentBlock $createContentBlock)
     {
-        $createContentBlock->extraId = $this->getNewExtraId($createContentBlock->title);
+        $createContentBlock->extraId = $this->widget->getNewExtraId($createContentBlock);
```

Comment:
I would keep this a local method, no need for a separate service for this and emediatly set the settings as well, you can pass a settingsBag with the values in an array passed to the constructor